test-lit/lit_port.py

```diff
@@ -0,0 +1,256 @@
+""" lit_port.py
+
+    Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+    See https://llvm.org/LICENSE.txt for license information.
+    SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+
+import argparse
+import os
+import re
+import shlex
+import sys
+import subprocess
+from pathlib import Path
+
+from test_splitter import TestSplitter
+
+ROOT = Path.cwd()
+while ROOT.name != "f18":
```

Comment:
can this f18 name be a global variable since in future this could also be flang. 
Ignore if this is only meant to be a run once script.

Reply:
After all tests are lit compatible, this script will be obsolete. The question is how likely is F18 to be renamed between now and then. I imagine while flang-compiler/flang is active, this won't happen, so we're probably okay for now.